<a href="https://colab.research.google.com/github/svohra10/Project-2-Coding-Dojo/blob/main/Project_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [3]:
# Imports
import pandas as pd
import numpy as np
from google.colab import drive
drive.mount('/content/drive')


Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


# Data Set 1


1.   Kaggel 
2.   Traffic congestion measurements across 65 road ways. 


roadways. 

In [5]:
#real data set is commented out due to size
df = pd.read_csv('/content/drive/MyDrive/Part-time Data Science - Python Basics/traffic.csv')
df.head()

row_id                 time  x  y direction  congestion
0       0  1991-04-01 00:00:00  0  0        EB          70
1       1  1991-04-01 00:00:00  0  0        NB          49
2       2  1991-04-01 00:00:00  0  0        SB          24
3       3  1991-04-01 00:00:00  0  1        EB          18
4       4  1991-04-01 00:00:00  0  1        NB          60

3) What is the target?
Congestion 

4) Is this a classification or regression problem?
Regression 


5) How many features?

4

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 848835 entries, 0 to 848834
Data columns (total 6 columns):
 #   Column      Non-Null Count   Dtype 
---  ------      --------------   ----- 
 0   row_id      848835 non-null  int64 
 1   time        848835 non-null  object
 2   x           848835 non-null  int64 
 3   y           848835 non-null  int64 
 4   direction   848835 non-null  object
 5   congestion  848835 non-null  int64 
dtypes: int64(4), object(2)
memory usage: 38.9+ MB


6) How many rows of data.

848834

7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?

I feel like i will run into issues converting the time to be regressed. Also I feel like there will be issues with the x and y coordinates. 

# Data Set 2 

1.   Kaggel
2.   housing data to predict final price of each home based on 79 variables



3) What is the target?
Final house price  

In [7]:
df = pd.read_csv('/content/drive/MyDrive/Part-time Data Science - Python Basics/hosue_ train.csv')
df.head()

Id  MSSubClass MSZoning  LotFrontage  LotArea Street Alley LotShape  \
0   1          60       RL         65.0     8450   Pave   NaN      Reg   
1   2          20       RL         80.0     9600   Pave   NaN      Reg   
2   3          60       RL         68.0    11250   Pave   NaN      IR1   
3   4          70       RL         60.0     9550   Pave   NaN      IR1   
4   5          60       RL         84.0    14260   Pave   NaN      IR1   

  LandContour Utilities  ... PoolArea PoolQC Fence MiscFeature MiscVal MoSold  \
0         Lvl    AllPub  ...        0    NaN   NaN         NaN       0      2   
1         Lvl    AllPub  ...        0    NaN   NaN         NaN       0      5   
2         Lvl    AllPub  ...        0    NaN   NaN         NaN       0      9   
3         Lvl    AllPub  ...        0    NaN   NaN         NaN       0      2   
4         Lvl    AllPub  ...        0    NaN   NaN         NaN       0     12   

  YrSold  SaleType  SaleCondition  SalePrice  
0   2008        WD         Normal     208500  
1   2007        WD         Normal     181500  
2   2008        WD         Normal     223500  
3   2006        WD        Abnorml     140000  
4   2008        WD         Normal     250000  

[5 rows x 81 columns]

In [8]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1460 entries, 0 to 1459
Data columns (total 81 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Id             1460 non-null   int64  
 1   MSSubClass     1460 non-null   int64  
 2   MSZoning       1460 non-null   object 
 3   LotFrontage    1201 non-null   float64
 4   LotArea        1460 non-null   int64  
 5   Street         1460 non-null   object 
 6   Alley          91 non-null     object 
 7   LotShape       1460 non-null   object 
 8   LandContour    1460 non-null   object 
 9   Utilities      1460 non-null   object 
 10  LotConfig      1460 non-null   object 
 11  LandSlope      1460 non-null   object 
 12  Neighborhood   1460 non-null   object 
 13  Condition1     1460 non-null   object 
 14  Condition2     1460 non-null   object 
 15  BldgType       1460 non-null   object 
 16  HouseStyle     1460 non-null   object 
 17  OverallQual    1460 non-null   int64  
 18  OverallC

4) Is this a classification or regression problem?
Regression 

5) How many features?

80

6) How many rows?
1460

7) A lot of missing data points and a lot of features to be comparing. 